src/CashuMint.ts

```diff
@@ -57,26 +70,63 @@ class CashuMint {
 	}
 
 	async split(splitPayload: SplitPayload): Promise<SplitResponse> {
-		const { data } = await axios.post<SplitResponse>(`${this.mintUrl}/split`, splitPayload);
-		return data;
+		try {
+			const { data } = await axios.post<SplitResponse>(`${this.mintUrl}/split`, splitPayload);
+			checkResponse(data);
+			if (!isObj(data) || !Array.isArray(data?.fst) || !Array.isArray(data?.snd)) {
+				throw new Error('bad response');
+			}
+			return data;
+		} catch (err) {
+			checkResponseError(err);
+			throw err;
+		}
 	}
 	async melt(meltPayload: MeltPayload): Promise<MeltResponse> {
-		const { data } = await axios.post<MeltResponse>(`${this.mintUrl}/melt`, meltPayload);
-		return data;
+		try {
+			const { data } = await axios.post<MeltResponse>(`${this.mintUrl}/melt`, meltPayload);
+			checkResponse(data);
+			checkResponse(data);
+			if (!isObj(data) || typeof data?.paid !== 'boolean' || typeof data?.preimage !== 'string') {
```

Comment:
since `change` is optional, we don't check it. I think that's ok, we could check it's format if it exists, but I don't think it is strictly necessary